parser.js

```diff
@@ -274,7 +274,7 @@
     function createClassRange(min, max, from, to) {
       // See 15.10.2.15:
       if (min.codePoint > max.codePoint) {
-        throw SyntaxError('invalid range in character class');
+        bail('invalid range in character class: ' + min.raw + '-' + max.raw, from, to);
```

Comment:
> Can make the change with or without escaping of the content, but personally I found unescaped without wrappers to be the clearest.

Good thoughts - I like and agree with them.

---

How about this: 

> invalid range in character class at position 1: a-- \n    [a--z]\n     ^

In this way, the position information is moved before the colon and then bit after the colon holds the detail information about what went wrong.